// react-d3-zoom/lib/scatter.js
"use strict";

Object.defineProperty(exports, '__esModule', {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ('value' in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

var _get = function get(_x, _x2, _x3) { var _again = true; _function: while (_again) { var object = _x, property = _x2, receiver = _x3; desc = parent = getter = undefined; _again = false; if (object === null) object = Function.prototype; var desc = Object.getOwnPropertyDescriptor(object, property); if (desc === undefined) { var parent = Object.getPrototypeOf(object); if (parent === null) { return undefined; } else { _x = parent; _x2 = property; _x3 = receiver; _again = true; continue _function; } } else if ('value' in desc) { return desc.value; } else { var getter = desc.get; if (getter === undefined) { return undefined; } return getter.call(receiver); } } };

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } }

function _inherits(subClass, superClass) { if (typeof superClass !== 'function' && superClass !== null) { throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactD3Core = require('react-d3-core');

var _reactD3Basic = require('react-d3-basic');

var _inheritIndex = require('./inherit/index');

var _inheritIndex2 = _interopRequireDefault(_inheritIndex);

var _utilsZoom_focus = require('./utils/zoom_focus');

var _utilsZoom_focus2 = _interopRequireDefault(_utilsZoom_focus);

var _commonProps = require('./commonProps');

var _commonProps2 = _interopRequireDefault(_commonProps);

var ScatterZoom = (function (_ZoomSet) {
  _inherits(ScatterZoom, _ZoomSet);

  function ScatterZoom(props) {
    _classCallCheck(this, ScatterZoom);

    _get(Object.getPrototypeOf(ScatterZoom.prototype), 'constructor', this).call(this, props);

    var _props = this.props;
    var margins = _props.margins;
    var width = _props.width;
    var height = _props.height;

    this.zoomed = this.zoomed.bind(this);
    this.mkXDomain();
    this.mkYDomain();

    this.state = {
      xDomainSet: this.setXDomain,
      yDomainSet: this.setYDomain,
      onZoom: this.zoomed,
      d3EventSet: null,
      xRange: this.props.xRange || [0, width - margins.left - margins.right],
      yRange: this.props.yRange || [height - margins.top - margins.bottom, 0],
      xRangeRoundBands: this.props.xRangeRoundBands || { interval: [0, width - margins.left - margins.right], padding: .1 },
      zoomType: 'scatter'
    };

    this.mkXScale(this.setXDomain);
    this.mkYScale(this.setYDomain);

    this.state = Object.assign(this.state, {
      xScaleSet: this.setXScale,
      yScaleSet: this.setYScale
    });
  }

  _createClass(ScatterZoom, [{
    key: 'render',
    value: function render() {
      var _state = this.state;
      var xDomainSet = _state.xDomainSet;
      var yDomainSet = _state.yDomainSet;

      var focus = _react2['default'].createElement(_utilsZoom_focus2['default'], this.props);

      return _react2['default'].createElement(
        'div',
        null,
        _react2['default'].createElement(
          _reactD3Core.Chart,
          _extends({}, this.props, this.state),
          _react2['default'].createElement(_reactD3Basic.ScatterPlot, _extends({}, this.props, this.state, { xDomain: xDomainSet, yDomain: yDomainSet, showZoom: true })),
          focus
        )
      );
    }
  }], [{
    key: 'defaultProps',
    value: _commonProps2['default'],
    enumerable: true
  }]);

  return ScatterZoom;
})(_inheritIndex2['default']);

exports['default'] = ScatterZoom;
module.exports = exports['default'];
